src/plugins/network/testmod_network.c

```diff
@@ -27,8 +32,53 @@ int main (int argc, char ** argv)
 	init (argc, argv);
 
 	testIPAll ();
+	testPorts ();
 
 	print_result ("testmod_network");
 
 	return nbError;
 }
+
+static void testPort (char const * const port, const int ret, char const * const version, char const * const metaName)
+{
+	Key * parentKey = keyNew ("user/tests/port", KEY_VALUE, "", KEY_END);
+	KeySet * conf = ksNew (0, KS_END);
+	KeySet * ks = ksNew (10, keyNew ("user/test/port/totest", KEY_VALUE, port, KEY_META, metaName, version, KEY_END), KS_END);
+	PLUGIN_OPEN (PLUGIN_NAME);
+	const int pluginStatus = plugin->kdbSet (plugin, ks, parentKey);
+	char message[200];
+	(void) snprintf (message, 200, "validation of %s “%s” returned %d instead of %d", version[0] == '\0' ? "Port" : version, port,
+			 pluginStatus, ret);
+	succeed_if (pluginStatus == ret, message);
+	ksDel (ks);
+	keyDel (parentKey);
+	PLUGIN_CLOSE ();
+}
+
+static inline void testPortAny (char const * const port, int ret)
+{
+	testPort (port, ret, "", "check/port");
+}
+
+static inline void testListenPortAny (char const * const port, int ret)
+{
+	testPort (port, ret, "", "check/port/listen");
+}
+
+static void testPorts (void)
+{
+	testPortAny ("0", 1);
+	testPortAny ("1234", 1);
+	testPortAny ("65535", 1);
+	testPortAny ("ssh", 1);
+	testPortAny ("https", 1);
+
+	testPortAny ("65536", -1);
+	testPortAny ("-1", -1);
+	testPortAny ("22d", -1);
+	testPortAny ("myInvalidServiceName", -1);
+
+	// These tests aren't portable I guess
+	// testListenPortAny ("http", -1);
+	testListenPortAny ("8080", 1);
```

Comment:
8080 is quite likely to be in use. There are some reserved ports, maybe let us use?

these https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers

Reply:
Do you have a suggestion/preference for which one to use?

Reply:
I tried nearly all ports from 1-100 and only http does work on my system (but not on the build system) when I try to test for unavaliable ports as root. So what should I do now?

You can give me ports which are reserved on the build server for sure or I cannot test it at all. (a -1 case at least)

Reply:
Did you consider that Ports below 1024 can only be used by root?

Reply:
Yes I did. I was able to use all ports and `netstat -tulpn` also only showed that port 80 was in use (of all ports below 1000). I am running the tests as root so I can try to bind to the port.

It seems like this is not testable because every build system/ (maybe even distro of linux) uses different ports or uses them just in a very short window (like port 7 `echo`) that it is not deterministic if a port is in use or not.

One possibility is to take the `ssh` port from the build server as it will most likely be in use all the time?